src/ucs/type/class.h

```diff
@@ -221,9 +221,15 @@ struct ucs_class {
                                       _argtype **obj_p)
 #define UCS_CLASS_DEFINE_NAMED_NEW_FUNC(_name, _type, _argtype, ...) \
     UCS_CLASS_DECLARE_NAMED_NEW_FUNC(_name, _argtype, ## __VA_ARGS__) { \
-        return UCS_CLASS_NEW(_type, obj_p \
-                             UCS_PP_FOREACH(_UCS_CLASS_INIT_ARG_PASS, _, \
-                                            UCS_PP_SEQ(UCS_PP_NUM_ARGS(__VA_ARGS__)))); \
+        ucs_status_t status; \
+        \
+        *obj_p = NULL; \
+        \
+        status = UCS_CLASS_NEW(_type, obj_p \
```

Comment:
maybe add it to UCS_CLASS_NEW macro? so it would fix issues not only when defining a wrapper function for UCS_CLASS_NEW, but also when UCS_CLASS_NEW used directly

Reply:
it was the first implementation (in `_UCS_CLASS_NEW`), Coverity reports warning in this case
Coverity thinks that a return value from `UCS_CLASS_NEW` macro can be `UCS_OK`, but then `_name()` (created from `UCS_CLASS_DECLARE_NAMED_NEW_FUNC`) can return `!UCS_OK`.
But the return value of `_name()` (created from `UCS_CLASS_DECLARE_NAMED_NEW_FUNC`) = the return value from `UCS_CLASS_NEW` macro